### PR TITLE
[LIT-2333] Add e2e test for initializing multiple clients

### PR DESCRIPTION
**Pull request by cairomassimo**
# Description

This PR enables the use of multiple instances of `LitNodeClient`, in order to connect to different networks in the same application, at the same time.

This was achieved by removing the global variables `litConfig`, `litNodeClient`, `logger` and `logManager` and their references.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added an e2e test which calls `.connect()` on multiple clients.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: e2e-nodejs/group-connection/test-connection-multi-network.mjs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      debug: globalThis.LitCI.debug,
+    });
+    log(`connecting to ${network.toUpperCase()}...`);
+    await client.connect();
+
+    // ==================== Post-Validation ====================
+    if (!client.ready) {
+      return fail('client not ready');
+    }
+    if (client.config.litNetwork !== network) {
```
Comment (joshLong145) [inline e2e-nodejs/group-connection/test-connection-multi-network.mjs:21]: Could we add a check on `globalThis` to assert on the new global namespacing?

File: packages/lit-auth-client/src/lib/lit-auth-client.ts [hunk 2]
```diff
@@ -60,7 +60,6 @@ export class LitAuthClient {
    */
   constructor(options?: LitAuthClientOptions) {
     this.providers = new Map();
-    bootstrapLogManager('auth-client');
```
Comment (joshLong145) [inline packages/lit-auth-client/src/lib/lit-auth-client.ts:63]: why did we remove this line?

File: packages/misc/src/lib/misc.ts [hunk 3]
```diff
@@ -27,10 +27,10 @@ import {
 } from '@lit-protocol/types';
 import { JsonRpcProvider } from '@ethersproject/providers';
 import { Contract } from '@ethersproject/contracts';
-import { LogLevel, LogManager } from '@lit-protocol/logger';
+import { LogManager } from '@lit-protocol/logger';
 import { version } from '@lit-protocol/constants';
 
-const logBuffer: Array<Array<any>> = [];
+const defaultLogger = LogManager.Instance.get('lit-sdk');
```
Comment (joshLong145) [inline packages/misc/src/lib/misc.ts:33]: will this be a global logger? I would like to keep the package names as categories if possible. Open to discussion however.

File: packages/misc/src/lib/misc.ts [hunk 4]
```diff
@@ -233,121 +211,19 @@ export const getLoggerbyId = (id: string) => {
...
-
-  // if there are there are logs in buffer, print them first and empty the buffer.
-  while (logBuffer.length > 0) {
-    const log = logBuffer.shift() ?? '';
-    globalThis?.logger &&
-      globalThis.logManager.get(globalThis.logger.category).error(...log);
-  }
-
-  globalThis?.logger &&
-    globalThis.logManager.get(globalThis.logger.category).error(...args);
```
Comment (joshLong145) [inline packages/misc/src/lib/misc.ts:350]: Why did we remove all of these utilities?

**Review by joshLong145 — CHANGES_REQUESTED** [review]
I see you removed methods which we where using in many packages. Can you outline the implications and behavior changes that are introduced with these utilities gone? #331 will also add many of these methods. back when merged. There does not seem to be a path forward for the above mentioned pr with the current implementation.

**Comment by cairomassimo** [comment]
@joshLong145 I updated the PR so that it _only_ adds the e2e test for connecting to multiple networks. The test passes with the code as is.
We discussed about the changes on the logger in a call, but overall I think it makes sense to not apply those changes in this PR anyway.

To be clear, with the code as it is, creating multiple clients does mess up the global state (`globalThis.litConfig` and `globalThis.litNodeClient` get replaced). However:
1. this is the current behavior anyway, so the PR is not introducing a regression here,
2. I think we should recommend the SDK consumers to _not rely on the global state_ anyway, rather than introducing a namespacing convention, espcially since they have to instantiate the clients themselves anyway.

There is one more check to do: making sure that there are no conflicts with the usage of `localStorage` when instantiating more than one client. I couldn't find any, but it's better that someone with more experience with the codebase has a look at this specifically.

File: e2e-nodejs/group-connection/test-connection-multi-network.mjs (hunk 1 above)
Comment (cairomassimo) [reply e2e-nodejs/group-connection/test-connection-multi-network.mjs:21]: I didn't implement the global namespacing, I think it's a risky approach (see also https://github.com/LIT-Protocol/js-sdk/pull/331#issuecomment-1916891635). There could be a use case for global namespaces I'm not aware of yet: in that case please let me know.

**Comment by joshLong145** [comment]
@cairomassimo can we close this pr since we no longer need to make code changes?

**Comment by cairomassimo** [comment]
> @cairomassimo can we close this pr since we no longer need to make code changes?

@joshLong145 It's not much, but we can merge it so we have the new e2e test case? I'm making it "ready for review" just in case

**Review by Ansonhkg — APPROVED** [review]
I'm ok with having this for our e2e tests

**Pull request merged**